nsxt/data_source_nsxt_logical_switch_test.go

```diff
@@ -0,0 +1,117 @@
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccDataSourceNsxtLogicalSwitchDeleteByName(switchName)
+		},
+		Steps: []resource.TestStep{
+			{
+				PreConfig: func() {
+					if err := testAccDataSourceNsxtLogicalSwitchCreate(switchName); err != nil {
+						panic(err)
+					}
+				},
+				Config: testAccNSXLogicalSwitchReadTemplate(switchName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(testResourceName, "display_name", switchName),
+					resource.TestCheckResourceAttr(testResourceName, "description", switchName),
+					resource.TestCheckResourceAttr(testResourceName, "transport_zone_id", transportZoneID),
+				),
+			},
+			{
+				Config: testAccNSXNoLogicalSwitchTemplate(),
+			},
+		},
+	})
+}
+
+func testAccDataSourceNsxtLogicalSwitchCreate(switchName string) error {
+	nsxClient, err := testAccGetClient()
+	if err != nil {
+		return fmt.Errorf("Error during test client initialization: %v", err)
+	}
+
+	displayName := switchName
+	description := switchName
+	adminState := "UP"
+	replicationMode := "MTEP"
+	transportZoneID := getTestTransportZoneID()
+
+	logicalSwitch := manager.LogicalSwitch{
+		AdminState:      adminState,
+		Description:     description,
+		DisplayName:     displayName,
+		TransportZoneId: transportZoneID,
+		ReplicationMode: replicationMode,
+	}
+
+	logicalSwitch, resp, err := nsxClient.LogicalSwitchingApi.CreateLogicalSwitch(nsxClient.Context, logicalSwitch)
+	if err != nil {
+		return fmt.Errorf("Error during logical switch creation: %v", err)
+	}
+
+	if resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("Unexpected status returned during logical switch creation: %v", resp.StatusCode)
+	}
+	return nil
+}
+
+func testAccDataSourceNsxtLogicalSwitchDeleteByName(switchName string) error {
+	nsxClient, err := testAccGetClient()
+	if err != nil {
+		return fmt.Errorf("Error during test client initialization: %v", err)
+	}
+
+	// Find the object by name
+	objList, _, err := nsxClient.LogicalSwitchingApi.ListLogicalSwitches(nsxClient.Context, nil)
+	if err != nil {
+		return fmt.Errorf("Error while reading logical switches: %v", err)
+	}
+	// go over the list to find the correct one
+	for _, objInList := range objList.Results {
+		if objInList.DisplayName == switchName {
+			localVarOptionals := make(map[string]interface{})
+			responseCode, err := nsxClient.LogicalSwitchingApi.DeleteLogicalSwitch(nsxClient.Context, objInList.Id, localVarOptionals)
+			if err != nil {
+				return fmt.Errorf("Error during logical switch deletion: %v", err)
+			}
+
+			if responseCode.StatusCode != http.StatusOK {
+				return fmt.Errorf("Unexpected status returned during logical switch deletion: %v", responseCode.StatusCode)
+			}
+			return nil
+		}
+	}
+	return fmt.Errorf("Error while deleting logical switch '%s': service not found", switchName)
+}
+
+func testAccNSXLogicalSwitchReadTemplate(name string) string {
+	return fmt.Sprintf(`
```

Comment:
May I suggest using logical switch resource to create the resource for this test? This way we won't need to introduce the new environment variable, and require less code.

Reply:
That way, changes on the resource_nsxt_logical_switch might possibly affect the data source's test and in some cases possibly require some changes on the data source's code itself.
If that's alright or even preferable, we can do that. Wdyt?

Reply:
I think its OK. We prefer to have as little pre-configuring as possible before we run acc tests, so not having to pre-create the LS would help. Thanks!

Reply:
Sorry for the really late reply!
When trying to do this, I can't find any example reference for using the resource in the data source test. Can you point me to any? All the other data sources seem to use the api directly to pre-create the related resource in the test.